pkgs/development/python-modules/jax/default.nix

```diff
@@ -17,97 +17,96 @@
 
 let
   usingMKL = blas.implementation == "mkl" || lapack.implementation == "mkl";
-in
-buildPythonPackage rec {
-  pname = "jax";
-  version = "0.3.6";
-  format = "setuptools";
 
-  disabled = pythonOlder "3.7";
+  jax = buildPythonPackage rec {
+    pname = "jax";
+    version = "0.3.16";
+    format = "setuptools";
 
-  src = fetchFromGitHub {
-    owner = "google";
-    repo = pname;
-    rev = "jax-v${version}";
-    hash = "sha256-eGdAEZFHadNTHgciP4KMYHdwksz9g6un0Ar+A/KV5TE=";
-  };
+    disabled = pythonOlder "3.7";
 
-  patches = [
-    # See https://github.com/google/jax/issues/7944
-    ./cache-fix.patch
+    src = fetchFromGitHub {
+      owner = "google";
+      repo = pname;
+      rev = "jax-v${version}";
+      hash = "sha256-4idh7boqBXSO9vEHxEcrzXjBIrKmmXiCf6cXh7En1/I=";
+    };
 
-    # See https://github.com/google/jax/issues/10292
-    (fetchpatch {
-      url = "https://github.com/google/jax/commit/cadc8046d56e0c1433cf48a2f106947d5f4ecbfd.patch";
-      hash = "sha256-jrpIqt4LzWAswt/Cpwtfa5d1Yn31HcXkVH3ETmaigA0=";
-    })
-  ];
+    # jaxlib is _not_ included in propagatedBuildInputs because there are
+    # different versions of jaxlib depending on the desired target hardware. The
+    # JAX project ships separate wheels for CPU, GPU, and TPU. Currently only the
+    # CPU wheel is packaged.
+    propagatedBuildInputs = [
+      absl-py
+      etils
+      numpy
+      opt-einsum
+      scipy
+      typing-extensions
+    ] ++ etils.optional-dependencies.epath;
 
-  # jaxlib is _not_ included in propagatedBuildInputs because there are
-  # different versions of jaxlib depending on the desired target hardware. The
-  # JAX project ships separate wheels for CPU, GPU, and TPU. Currently only the
-  # CPU wheel is packaged.
-  propagatedBuildInputs = [
-    absl-py
-    numpy
-    opt-einsum
-    scipy
-    typing-extensions
-  ];
+    # tests have dependency cycle: jax -> etils -> jax
+    doCheck = false;
```

Comment:
We should really keep the jax test suite running. That may mean breaking the dependency cycle at etils, but the JAX test suite is comprehensive and important to get right.

Reply:
Resolved (https://github.com/NixOS/nixpkgs/pull/183051#issuecomment-1224954563)